src/HomePage.jsx

```diff
@@ -3,8 +3,12 @@ import PropTypes from 'prop-types';
 
 import AdminDashboardPage from './pages/AdminDashboardPage';
 import VolunteerDashboardPage from './pages/VolunteerDashboardPage';
+import { useUserContext } from './common/UserContext/UserContext';
+
+const HomePage = ({ onAdminPortal }) => {
+  const userData = useUserContext();
+  const isAdmin = userData.userData.role === 'admin';
```

Comment:
For some reason, there's 2 home page files..... can you delete this one and use the one in `/pages`? These changes look good just move them over to the correct file